[Green CI] Fix test 01903_correct_block_size_prediction_with_default

pamarcos opened the pull request:
According to the [original issue](https://github.com/ClickHouse/ClickHouse/issues/17317), the test was introduced to check that the SQL expression in a DEFAULT column didn't consume more memory than using the SQL expression separately. With this in mind, rather than having an arbitrary hardcoded value of memory cap at 100MB, let's compare the memory usage for both statements.

I added `no-random-settings` because having randomization resulted in some variations in the memory usage that is not desirable for this test. e.g. the first query without any WHERE condition may consume 32MB instead of the usual 10MB.

### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ] <!---ci_set_required--> Allow: All Required Checks
- [x] <!---ci_include_stateless--> Allow: Stateless tests
- [ ] <!---ci_include_stateful--> Allow: Stateful tests
- [ ] <!---ci_include_integration--> Allow: Integration Tests
- [ ] <!---ci_include_performance--> Allow: Performance tests
- [ ] <!---ci_set_builds--> Allow: All Builds
- [ ] <!---batch_0_1--> Allow: batch 1, 2 for multi-batch jobs
- [ ] <!---batch_2_3--> Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ] <!---ci_exclude_style--> Exclude: Style check
- [ ] <!---ci_exclude_fast--> Exclude: Fast test
- [ ] <!---ci_exclude_asan--> Exclude: All with ASAN
- [ ] <!---ci_exclude_tsan|msan|ubsan|coverage--> Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ] <!---ci_exclude_aarch64|release|debug--> Exclude: All with aarch64, release, debug
---
- [ ] <!---do_not_test--> Do not test
- [ ] <!---woolen_wolfdog--> Woolen Wolfdog
- [ ] <!---upload_all--> Upload binaries for special builds
- [x] <!---no_merge_commit--> Disable merge-commit
- [ ] <!---no_ci_cache--> Disable CI cache

Closes https://github.com/ClickHouse/clickhouse-private/issues/14196

pamarcos committed:
Fix test 01903_correct_block_size_prediction_with_default

- Don't allow random settings that affect the memory usage
- Run two queries and compare the memory usage, rather than
  having an arbitrary hardcoded value

robot-ch-test-poll4 commented:
<!-- automatic status comment for PR #68203 from pamarcos/ClickHouse:fix-test-01903_correct_block_size_prediction_with_default -->
*This is an automated comment for commit 06ceaee50218507f49bfc714903240ad4b5d81a0 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68203/06ceaee50218507f49bfc714903240ad4b5d81a0/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68161/d314e5aa45fb8ac91324721ab278185b09437a40/docs_check.html">✅ success</a></td></tr>
<tr><td>Flaky tests</td><td>Checks if new added or modified tests are flaky by running them repeatedly, in parallel, with more randomization. Functional tests are run 100 times with address sanitizer, and additional randomization of thread scheduling. Integration tests are run up to 10 times. If at least once a new test has failed, or was too long, this check will be red. We don't allow flaky tests, read <a href="https://clickhouse.com/blog/decorating-a-christmas-tree-with-the-help-of-flaky-tests/">the doc</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67964/ee433684ddc4614a5bb93dbfd1e3b481a2f343d6/integration_tests_flaky_check__asan_.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0aa30b10d5b4a33c7b088bfcaeaeff2620ef73ec/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

pamarcos committed:
Small fix to filter by current_database in system.query_log

pamarcos committed:
Fix indent

divanik commented on tests/queries/0_stateless/01903_correct_block_size_prediction_with_default.sh:
```diff
@@ -0,0 +1,36 @@
+        SYSTEM FLUSH LOGS;
+        WITH memory_1 AS (SELECT memory_usage FROM system.query_log WHERE current_database = currentDatabase() AND query_id='$uuid_1' AND type = 'QueryFinish' as memory_1),
+             memory_2 AS (SELECT memory_usage FROM system.query_log WHERE current_database = currentDatabase() AND query_id='$uuid_2' AND type = 'QueryFinish' as memory_2)
+                SELECT memory_1.memory_usage <= 1.2 * memory_2.memory_usage FROM memory_1, memory_2;"
```
Add a condition that "memory_2.memory_usage <= 1.2 * memory_1.memory_usage", please

pamarcos replied:
Done in https://github.com/ClickHouse/ClickHouse/pull/68203/commits/858b7e55d0df3db1412d538f701c30584b5783bf

pamarcos committed:
Improve condition in case the default column consumes slightly more memory

It never happened in the few hundreds of tests I ran successfully,
but we'd rather be safe than sorry.
